### PR TITLE
fix(docs-infra): skip navigation to card if user clicks on anchor

### LINKED ISSUE #57078: Links embedded inside docs sidebar popouts are not clickable
### Describe the problem that you experienced

Whenever you click on the sidebar popout elements on the new docs site, this has the effect of scrolling to the top of the sidebar menu's content.

This is problematic in two ways: it's unintuitive, as there is no obvious reason why clicking on an item half way down the list should revert the scroll, and it also seems to prevent default link behaviour on any anchor tags inside this panel.

### Enter the URL of the topic with the problem

https://angular.dev/cli/build#Arguments

### Describe what you were looking for in the documentation

Want to click links in the side menu

### Describe the actions that led you to experience the problem

Try clicking on the link in the documentation of the `source-map` argument:

> Output source maps for scripts and styles. For more information, see https://angular.dev/reference/configs/workspace-config#source-map-configuration.

### Describe what you want to experience that would fix the problem

Should be possible to navigate to the new page

### Add a screenshot if that helps illustrate the problem

image

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true
Chrome 126 on Windows
```

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by 9kubczas4**
fix(docs-infra): skip navigation to card if user clicks on anchor

If user clicks on anchor inside member card (Reference) then skip navigation to card section and allow to navigate to anchor link.

**Changed functionality: skip scroll when user clicks on the card body. Scroll only after click on the header of the card.**

PR fixes: https://github.com/angular/angular/issues/57078.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
If user clicks on link which is inside the member card, then instead of navigate to the page from this link there is navigation to the member card section.

image

Issue Number: N/A

## What is the new behavior?

Navigation to the page from the link is done.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

**Review by devversion — APPROVED** [review]

**Review by JeanMeche — APPROVED** [review]

File: adev/src/app/features/references/services/reference-scroll-handler.service.ts [hunk 1]
```diff
@@ -114,7 +114,10 @@ export class ReferenceScrollHandler implements OnDestroy, ReferenceScrollHandler
     this.getAllMemberCards().forEach((card) => {
       this.cardOffsetTop.set(card.id, card.offsetTop);
       fromEvent(card, 'click')
-        .pipe(takeUntilDestroyed(this.destroyRef))
+        .pipe(
+          skipWhile((event) => event.target instanceof HTMLAnchorElement),
```
Comment (JoostK) [inline adev/src/app/features/references/services/reference-scroll-handler.service.ts:118]: `skipWhile` doesn't get you the semantics we're after here; we want to exclude _all_ anchor clicks, not just the first ones until a non-anchor click occurs. Consequently, this should be

```
          filter((event) => !(event.target instanceof HTMLAnchorElement)),
```

**Review by JoostK — CHANGES_REQUESTED** [review]

File: adev/src/app/features/references/services/reference-scroll-handler.service.ts (hunk 1 above)
Comment (JeanMeche) [reply adev/src/app/features/references/services/reference-scroll-handler.service.ts:118]: Actually there is a much bigger issue. 
This service is provided at the app level, `DestroyRef` never fires, the listeners leak.

Comment (9kubczas4) [reply adev/src/app/features/references/services/reference-scroll-handler.service.ts]: Thank you for the good catches.
In the recent commit, I've used the right RxJs operator and improved DI.

**Comment by benelliott** [comment]
Thanks @9kubczas4 for writing a fix so quickly! I was wondering whether you think it's correct for a click on that card to reset the scroll position, regardless of if the target was an anchor element? In other words, if you click on the textual part of the card description (e.g. to select a section for copy-paste), it resets the scroll position, which feels odd to me.

File: adev/src/app/features/references/services/reference-scroll-handler.service.ts [hunk 2]
```diff
@@ -56,6 +53,10 @@ export class ReferenceScrollHandler implements OnDestroy, ReferenceScrollHandler
   }
 
   setupListeners(tocSelector: string): void {
+    if (!this.isBrowser) {
```
[the remark this answers is not in the document]
Comment (9kubczas4) [reply adev/src/app/features/references/services/reference-scroll-handler.service.ts:56]: The problem is that `setupListeners` is called when `contentLoaded` from `docs-viewer` will emit value.

```html
<docs-viewer
  class="adev-cli-members-container"
  [docContent]="cardsInnerHtml()"
  (contentLoaded)="contentLoaded()"
/>
```

```ts
  contentLoaded(): void {
    this.scrollHandler.setupListeners(CLI_TOC);
    this.scrollHandler.updateMembersMarginTop(CLI_TOC);
  }
```

File: adev/src/app/features/references/services/reference-scroll-handler.service.ts [hunk 3]
```diff
@@ -6,45 +6,42 @@
...
   OnDestroy,
+  PLATFORM_ID,
   afterNextRender,
   inject,
   signal,
 } from '@angular/core';
 import {takeUntilDestroyed} from '@angular/core/rxjs-interop';
 import {fromEvent} from 'rxjs';
-import {auditTime} from 'rxjs/operators';
+import {auditTime, filter} from 'rxjs/operators';
```
Comment (JoostK) [inline adev/src/app/features/references/services/reference-scroll-handler.service.ts:22]: I think this can be dropped again

**Review by JoostK — APPROVED** [review]

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 12eb0053d908a6d540e36c3a0269568f96a1554f.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**